extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java

```diff
@@ -737,26 +735,41 @@ public void onFetchPage(@Nonnull final Downloader downloader)
             youtubePlayerResponse = playerResponse;
         }
 
-        JsonObject playabilityStatus = (playerResponse == null ? youtubePlayerResponse : playerResponse)
-                .getObject("playabilityStatus");
+        final JsonObject playabilityStatus = (playerResponse == null ? youtubePlayerResponse
+                : playerResponse).getObject("playabilityStatus");
+
+        checkPlayabilityStatus(youtubePlayerResponse, playabilityStatus);
+
+        nextResponse = getJsonPostResponse("next", body, localization);
+
+        streamingData = playerResponse.getObject("streamingData");
+        if (hasOtfStreams() || isCipherProtectedContent()) {
+            fetchAndroidMobileJsonPlayer(contentCountry, localization, videoId, stsKnown);
```

Comment:
... when you can just do this?
```suggestion
            fetchAndroidMobileJsonPlayer(contentCountry, localization, videoId, sts != null);
```

Reply:
I used what you said in https://github.com/TeamNewPipe/NewPipeExtractor/pull/604#discussion_r646097890, which is better.